test/autorest/errorsgroup/zz_generated_models.go

```diff
@@ -13,63 +13,6 @@ type Animal struct {
 	AniType *string `json:"aniType,omitempty"`
 }
 
-type AnimalNotFound struct {
-	// REQUIRED
-	WhatNotFound *string `json:"whatNotFound,omitempty"`
-	Name         *string `json:"name,omitempty"`
-	Reason       *string `json:"reason,omitempty"`
-	SomeBaseProp *string `json:"someBaseProp,omitempty"`
-}
-
-// GetNotFoundErrorBase implements the NotFoundErrorBaseClassification interface for type AnimalNotFound.
-func (a *AnimalNotFound) GetNotFoundErrorBase() *NotFoundErrorBase {
-	return &NotFoundErrorBase{
-		Reason:       a.Reason,
-		WhatNotFound: a.WhatNotFound,
-		SomeBaseProp: a.SomeBaseProp,
-	}
-}
-
-type BaseError struct {
-	SomeBaseProp *string `json:"someBaseProp,omitempty"`
-}
-
-type LinkNotFound struct {
-	// REQUIRED
-	WhatNotFound   *string `json:"whatNotFound,omitempty"`
-	Reason         *string `json:"reason,omitempty"`
-	SomeBaseProp   *string `json:"someBaseProp,omitempty"`
-	WhatSubAddress *string `json:"whatSubAddress,omitempty"`
-}
-
-// GetNotFoundErrorBase implements the NotFoundErrorBaseClassification interface for type LinkNotFound.
-func (l *LinkNotFound) GetNotFoundErrorBase() *NotFoundErrorBase {
-	return &NotFoundErrorBase{
-		Reason:       l.Reason,
-		WhatNotFound: l.WhatNotFound,
-		SomeBaseProp: l.SomeBaseProp,
-	}
-}
-
-// NotFoundErrorBaseClassification provides polymorphic access to related types.
-// Call the interface's GetNotFoundErrorBase() method to access the common type.
-// Use a type switch to determine the concrete type.  The possible types are:
-// - *AnimalNotFound, *LinkNotFound, *NotFoundErrorBase
-type NotFoundErrorBaseClassification interface {
```

Comment:
We cannot remove a discriminator or any of its child types.  When encountering a discriminated type, you'll need to mark its entire hierarchy as required.

https://github.com/Azure/autorest.testserver/blob/main/swagger/xms-error-responses.json#L148

Reply:
All the error types have been removed because we do not use the error type in our SDK directly. If you think we should reserve them, I can change the logic. For all the discriminators, all the type chain will be reserved if one type has been referenced.